src/main/java/com/ngs/stash/externalhooks/hook/ExternalMergeCheckHook.java

```diff
@@ -139,27 +228,123 @@ public RepositoryHookResult preUpdate(PreRepositoryHookContext context, PullRequ
         if (!this.isLicenseValid()) {
             return RepositoryHookResult.rejected(
                 "License is not valid.",
-                "License for External Hooks Plugin is expired.\n"+
-                "Visit \"Manage add-ons\" page in your Bitbucket instance for more info."
+                "License for External Hooks Plugin is expired.\n" +
+                    "Visit \"Manage add-ons\" page in your Bitbucket instance for more info."
             );
         }
 
         try {
-            return runExternalHooks(pb, refChanges, summaryMsg);
+            final RepositoryHookResult result = runExternalHooks(pb, refChanges, summaryMsg);
+            check.setWasAccepted(result.isAccepted());
+            if (result.isRejected()) {
+                check.setLastExceptionSummary(result.getVetoes().get(0).getSummaryMessage());
+                check.setLastExceptionDetail(result.getVetoes().get(0).getDetailedMessage());
+            }
+            check.save();
+            return result;
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
             String detailedMsg = "Interrupted";
+            check.setWasAccepted(false);
+            check.setLastExceptionDetail(detailedMsg);
+            check.setLastExceptionSummary(summaryMsg);
+            check.save();
             return RepositoryHookResult.rejected(summaryMsg, detailedMsg);
         } catch (IOException e) {
-            log.log(SEVERE, "Error running "+exe+" in "+repoPath, e);
+            log.log(SEVERE, "Error running " + exe + " in " + repoPath, e);
             String detailedMsg = "I/O Error";
+            check.setWasAccepted(false);
+            check.setLastExceptionDetail(detailedMsg);
+            check.setLastExceptionSummary(summaryMsg);
+            check.save();
             return RepositoryHookResult.rejected(summaryMsg, detailedMsg);
         }
     }
 
+    @Override
+    public void onEnd(@Nonnull PreRepositoryHookContext context,
+                      @Nonnull PullRequestMergeHookRequest request,
+                      @Nonnull RepositoryHookResult result)
+    {
+        final PullRequest pr = request.getPullRequest();
+        final Settings settings = context.getSettings();
+
+        final Optional<PullRequestCheck> lastCheck = getCheck(pr);
+
+        if (!lastCheck.isPresent()) {
+            // this method was called without preUpdate first. This is unexpected.
+            return;
+        }
+
+        if (lastCheck.get().getVersion() == pr.getVersion() && lastCheck.get().getWasHandled()) {
+            return;
+        }
+
+        log.log(FINEST, "Adding new comment");
+
+        final StringBuffer comment = new StringBuffer();
+
+        if (result.isRejected()) {
+
+            result.getVetoes().forEach(repositoryHookVeto -> comment.append(String.format("%s (%s)",
```

Comment:
Can you please change formatting to support multiline output from hook?

Format `Merge request failed (...)` does not look when merge check script returns multiline message.

Reply:
What do you mean here? Translate \n to <br/>?

Reply:
No, I mean change format to `%s\n%s` if `getDetailedMessage()` is not empty and just `%s` otherwise. It seems that Bitbucket translates `\n` to `<br/>` automatically.